tests/jeos/firstrun.pm

```diff
@@ -72,19 +72,20 @@ sub run {
 
     my $lang = is_sle('15+') ? 'en_US' : get_var('JEOSINSTLANG', 'en_US');
 
-    # For 'en_US' pick 'en_US', for 'de_DE' select 'de_DE'
-    my %locale_key = ('en_US' => 'e', 'de_DE' => 'd');
     # For 'en_US' pick 'us', for 'de_DE' select 'de'
     my %keylayout_key = ('en_US' => 'u', 'de_DE' => 'd');
     # For 'en_US' pick 'UTC', for 'de_DE' select 'Europe/Berlin'
     my %tz_key = ('en_US' => 'u', 'de_DE' => 'e');
 
-    # Select locale
-    assert_screen 'jeos-locale', 300;
-    # Without this 'ret' sometimes won't get to the dialog
-    wait_still_screen;
-    send_key_until_needlematch "jeos-system-locale-$lang", $locale_key{$lang}, 50;
-    send_key 'ret';
+    # locale warning about missing glibc-locale package
+    if (is_sle('<=15-SP2') || is_leap('<=15.2')) {
+        assert_screen('jeos-locale', 300);
+        # Without this 'ret' sometimes won't get to the dialog
+        wait_still_screen;
+        send_key 'ret';
+    } else {
```

Comment:
Why `else` and why `jeos-firstboot-init` instead of `jeos-keylayout` or something like that?

Reply:
I would prefer to avoid usage of the same needle in `send_key_until_needlematch` call. Is it actually necessary to use `send_key_until_needlematch` over here as the `default=us` layout option should be always preselected?

Reply:
They would be different needles. send_key_until_needlematch is needed for other keyboard layouts